### PR TITLE
New upgrade.php

### PR DESCRIPTION
**Pull request by stefaweb**
convert to innoDB
new fields
new indexes
new look
delete spamscores table and index

```
MailWatch for MailScanner Database Upgrade

Testing connectivity to the database ................................. OK

Updating database schema:

 - Convert database to utf8........................................... ALREADY DONE

 - Drop `geoip_country` table......................................... ALREADY DROPPED
 - Drop `spamscores` table............................................ ALREADY DROPPED
 - Add autorelease table to `mailscanner` database.................... ALREADY EXIST

 - Fix schema for username field in `audit_log` table................. OK
 - Fix schema for id field in `blacklist` table....................... OK
 - Fix schema for username field in `users` table..................... OK
 - Fix schema for username field in `user_filters` table.............. OK
 - Fix schema for username field in `whitelist` table................. OK
 - Fix schema for password field in `users` table..................... OK
 - Fix schema for fullname field in `users` table..................... OK
 - Fix schema for rule_desc field in `mcp_rules` table................ OK

 - Add maillog_id field and primary key to `audit_log` table.......... ALREADY DONE
 - Add inq_id field and primary key to `inq` table.................... ALREADY DONE
 - Add maillog_id field and primary key to `maillog` table............ ALREADY DONE
 - Add mtalog_id field and primary key to `mtalog` table.............. ALREADY DONE
 - Add mtalog_id field and primary key to `outq` table................ ALREADY DONE
 - Add id field and primary key to `saved_filters` table.............. ALREADY DONE
 - Add mtalog_id field and primary key to `user_filters` table........ ALREADY DONE

 - Convert database to utf8mb4........................................ ALREADY DONE

 - Convert table `audit_log` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `autorelease` to utf8mb4............................. ALREADY CONVERTED
 - Convert table `blacklist` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `inq` to utf8mb4..................................... ALREADY CONVERTED
 - Convert table `maillog` to utf8mb4................................. ALREADY CONVERTED
 - Convert table `mcp_rules` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `mtalog` to utf8mb4.................................. ALREADY CONVERTED
 - Convert table `mtalog_ids` to utf8mb4.............................. ALREADY CONVERTED
 - Convert table `outq` to utf8mb4.................................... ALREADY CONVERTED
 - Convert table `saved_filters` to utf8mb4........................... ALREADY CONVERTED
 - Convert table `sa_rules` to utf8mb4................................ ALREADY CONVERTED
 - Convert table `users` to utf8mb4................................... ALREADY CONVERTED
 - Convert table `user_filters` to utf8mb4............................ ALREADY CONVERTED
 - Convert table `whitelist` to utf8mb4............................... ALREADY CONVERTED

 - Convert table `audit_log` to InnoDB................................ ALREADY CONVERTED
 - Convert table `autorelease` to InnoDB.............................. ALREADY CONVERTED
 - Convert table `blacklist` to InnoDB................................ ALREADY CONVERTED
 - Convert table `inq` to InnoDB...................................... ALREADY CONVERTED
 - Convert table `maillog` to InnoDB.................................. ALREADY CONVERTED
 - Convert table `mcp_rules` to InnoDB................................ ALREADY CONVERTED
 - Convert table `mtalog` to InnoDB................................... ALREADY CONVERTED
 - Convert table `mtalog_ids` to InnoDB............................... ALREADY CONVERTED
 - Convert table `outq` to InnoDB..................................... ALREADY CONVERTED
 - Convert table `saved_filters` to InnoDB............................ ALREADY CONVERTED
 - Convert table `sa_rules` to InnoDB................................. ALREADY CONVERTED
 - Convert table `users` to InnoDB.................................... ALREADY CONVERTED
 - Convert table `user_filters` to InnoDB............................. ALREADY CONVERTED
 - Convert table `whitelist` to InnoDB................................ ALREADY CONVERTED

 - Search for missing indexes......................................... DONE

Checking MailScanner.conf settings:

 - QuarantineWholeMessage ............................................ OK
 - QuarantineWholeMessagesAsQueueFiles ............................... OK
 - DetailedSpamReport ................................................ OK
 - IncludeScoresInSpamAssassinReport ................................. OK
 - SpamActions ....................................................... OK
 - HighScoringSpamActions ............................................ WARNING
 - AlwaysLookedUpLast ................................................ OK

Checking conf.php configuration entry:

 - All needed entries are OK
 - All obsolete entries are already removed

*** ERROR/WARNING SUMMARY ***
MailScanner.conf: HighScoringSpamActions != store (=delete)
```

### CODE REVIEW

**Comment by stefaweb** [comment]
Do you have a piece of code to insert separator?

```
- Convert table `users` to utf8mb4................................... ALREADY CONVERTED
 - Convert table `user_filters` to utf8mb4............................ ALREADY CONVERTED

 - Convert table `whitelist` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `audit_log` to innoDB................................ ALREADY CONVERTED
 - Convert table `autorelease` to innoDB.............................. ALREADY CONVERTED
```

The list is becoming long. I need to add separator in many place.

**Comment by asuweb** [comment]
Can't you just `echo "\n";`

**Comment by stefaweb** [comment]
Yes, I found  echo PHP_EOL;

**Comment by asuweb** [comment]
Yes, you can use that - that's what seems to be being used on the other lines.

**Comment by stefaweb** [comment]
New presentation

Will do index later.

**Comment by endelwar** [comment]
That's good, keep going :)

**Comment by stefaweb** [comment]
Please verify attentively.

I've a problem here:

` - Search for missing indexes.........................................`

Missing OK (alway OK)

**Comment by stefaweb** [comment]
Ready to merge.

**Comment by stefaweb** [comment]
Ready to merge.

**Comment by stefaweb** [comment]
Ready to merge.

**Comment by stefaweb** [comment]
Need to modify create.sql.

File: upgrade.php [hunk 1]
```diff
@@ -214,36 +214,46 @@ function getTableIndexes($table)
...
-
-    // Table audit_log
-    echo pad(' - Fix schema for username field in `audit_log` table');
-    $sql = "ALTER TABLE `audit_log` CHANGE `user` `user` VARCHAR( 255 ) NOT NULL DEFAULT ''";
+    // Table whitelist
+    echo pad(' - Fix schema for username field in `whitelist` table');
+    $sql = "ALTER TABLE `whitelist` CHANGE `id` `id` bigint(11) UNSIGNED NOT NULL AUTO_INCREMENT";
     executeQuery($sql);
 
+    // Revert back some tables to the right values due to previous errors in upgrade.php
```
Comment (stefaweb) [inline upgrade.php]: many people "dig" in develop... ;)

File: upgrade.php [hunk 2]
```diff
@@ -163,7 +163,7 @@ function getTableIndexes($table)
 // Test connectivity to the database
 
 echo PHP_EOL;
-echo 'MailWatch for MailScanner Upgrade' .  PHP_EOL;
+echo 'MailWatch for MailScanner Upgrade to 1.2 (RC5-dev)' .  PHP_EOL;
```
Comment (stefaweb) [inline upgrade.php]: Not sure to have done what you want.

File: upgrade.php (hunk 1 above)
Comment (stefaweb) [inline upgrade.php]: but these changes are from RC4 errors only.

File: upgrade.php [hunk 3]
```diff
@@ -34,8 +34,8 @@
     header('Content-type: text/plain');
 }
 
-$pathToFunctions = __DIR__ . '/mailscanner/functions.php';
-//$pathToFunctions = '/var/www/html/mailscanner/functions.php';
+// Edit if you changed webapp directory from default
+$pathToFunctions = '/var/www/html/mailscanner/functions.php';
```
Comment (stefaweb) [inline upgrade.php]: ok

File: upgrade.php [hunk 4]
```diff
@@ -288,13 +388,21 @@ function getTableIndexes($table)
...
-    if (false === check_table_exists('geoip_country')) {
-        echo ' ALREADY DROPPED' . PHP_EOL;
-    } else {
-        $sql = 'DROP TABLE IF EXISTS `geoip_country`';
-        executeQuery($sql);
+    echo PHP_EOL;
+
+    // Convert tables to innoDB using $utf8_tables array
+    foreach ($utf8_tables as $table) {
+        echo pad(' - Convert table `' . $table . '` to innoDB');
```
Comment (stefaweb) [inline upgrade.php]: ok

File: upgrade.php [hunk 5]
```diff
@@ -315,6 +423,9 @@ function getTableIndexes($table)
     );
 
     foreach ($indexes as $table => $indexlist) {
+        echo PHP_EOL;
+        echo ' - Search for missing indexes......................................... OK';
```
Comment (stefaweb) [inline upgrade.php]: ok

File: upgrade.php (hunk 1 above)
Comment (stefaweb) [inline upgrade.php]: ok

**Pull request merged**